cython_lint/cython_lint.py

```diff
@@ -836,6 +843,39 @@ def traverse(tree: ModuleNode) -> Iterator[NodeParent]:
         yield node_parent
 
 
+def _get_config(paths: list[pathlib.Path]) -> dict[str, Any]:
+    """Get the configuration from a config file
+
+    Search for a pyproject.toml or a setup.cfg file in common
+    parent directories of the given list of paths.
+    """
+    paths = [path.resolve() for path in paths]
+    root = pathlib.Path(os.path.commonpath(paths))
+    root = root.parent if root.is_file() else root
+
+    while root != root.parent:
+
+        # Look for pyproject.toml first
+        config_file = root / 'pyproject.toml'
+        if config_file.is_file():
+            config = tomllib.loads(config_file.read_text())
+            config = config.get('tool', {}).get('cython-lint', {})
+            if config:
+                return config
```

Comment:
does this logic come from some other tool? if so, it would be good to reference it. if not, we should add more extensive testing

Reply:
I tested the behavior of other linters like black and flake8. You don't have to run the tool from the root of your project and so they'll try to locate it by climbing up the path. I couldn't copy paste their logic because it's entangled in a much more complicated logic but I tried to mimic the behavior. I extended one of the tests in this regard.

Reply:
makes sense, thanks!